src/avt/Database/Ghost/avtUnstructuredDomainBoundaries.C

```diff
@@ -746,115 +995,21 @@ avtUnstructuredDomainBoundaries::ExchangeVector(vector<int> domainNum, bool isPo
     switch (maxDataType)
     {
         case VTK_FLOAT:
-            return ExchangeFloatVector(domainNum, isPointData, vectors);
+            return ExchangeData<float>(domainNum, isPointData, vectors);
             break;
         case VTK_DOUBLE:
-            return ExchangeDoubleVector(domainNum, isPointData, vectors);
+            return ExchangeData<double>(domainNum, isPointData, vectors);
             break;
         case VTK_INT:
         case VTK_UNSIGNED_INT:
-            return ExchangeIntVector(domainNum, isPointData, vectors);
+            return ExchangeData<int>(domainNum, isPointData, vectors);
```

Comment:
Why are `VTK_INT` and `VTK_UNSIGNED_INT` types handled by the same `<int>` block here?

Reply:
I thought that was strange too. But I didn't change functionality from the original code. Do you think that is a mistake?

Reply:
Ultimately, I think the decision to do it this way was based on the fact that the only relevant detail about the `ExchangeData` method is the size of the type and we know `int` and `unsigned int` will be the same size but cannot always be sure about `float`, `double` relative to each other or `int`.

Now, I don't honestly know but we may ultimate call an `MPI_XXX` method passing an equivalent `MPI_Type` (e.g. `MPI_INT`) and that is potentially more relevant...especially if we ever ran in a *heterogenous* context where different MPI ranks may have different meanings of `MPI_INT`. Honestly I don't know how MPI library itself might deal with that.

I think leaving as is is fine for now.

Reply:
ok